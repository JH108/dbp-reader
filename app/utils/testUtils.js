const getStartingEmAttributes = (color) =>
	`class="text-highlighted" style="background:linear-gradient(rgba(${color}),rgba(${color}))"`;

export { getStartingEmAttributes };
