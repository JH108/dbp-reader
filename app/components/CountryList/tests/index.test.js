import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { countries } from '../../../utils/testUtils/countryData';
import CountryList from '..';

const props = {
	countries: fromJS(countries),
	setCountryName: jest.fn(),
	toggleLanguageList: jest.fn(),
	setCountryListState: jest.fn(),
	getCountry: jest.fn(),
	getCountries: jest.fn(),
	filterText: '',
	active: true,
	loadingCountries: false,
	activeCountryName: 'ANY',
};

describe('CountryList component', () => {
	it('should match snapshot of active list', () => {
		const tree = renderer.create(<CountryList {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
