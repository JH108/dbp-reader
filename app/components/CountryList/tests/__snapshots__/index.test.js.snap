// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryList component should match snapshot of active list 1`] = `
<div
  className="text-selection-section"
>
  <div
    className="country-name-list"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="pull-down-refresh"
      style={
        Object {
          "height": 0,
          "maxHeight": 0,
        }
      }
    >
      <span
        style={
          Object {
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        Pull to Refresh
      </span>
    </div>
    <div
      style={
        Object {
          "height": 0,
          "overflow": "visible",
          "width": 0,
        }
      }
    >
      <div
        aria-label="grid"
        aria-readonly={true}
        className="ReactVirtualized__Grid ReactVirtualized__List"
        id="list-element"
        onScroll={[Function]}
        role="grid"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "boxSizing": "border-box",
            "direction": "ltr",
            "height": 0,
            "overflowX": "hidden",
            "overflowY": "auto",
            "position": "relative",
            "width": 0,
            "willChange": "transform",
          }
        }
        tabIndex={0}
      />
    </div>
  </div>
</div>
`;
